Buildings/Applications/DHC/EnergyTransferStations/CoolingIndirect.mo

```diff
@@ -0,0 +1,287 @@
+within Buildings.Applications.DHC.EnergyTransferStations;
+model CoolingIndirect
+  "Indirect cooling energy transfer station for district energy systems"
+  extends Buildings.Fluid.Interfaces.PartialFourPort(
+    redeclare package Medium1 = Medium,
+    redeclare package Medium2 = Medium);
+
+  package Medium = Buildings.Media.Water;
+
+  parameter Modelica.SIunits.SpecificHeatCapacity cp=
+    Medium.specificHeatCapacityCp(
+      Medium.setState_pTX(Medium.p_default, Medium.T_default, Medium.X_default))
+    "Default specific heat capacity of medium";
+
+  // mass flow rates
+  parameter Modelica.SIunits.MassFlowRate m1_flow_nominal(min=0,start=0.5)
+    "Nominal mass flow rate of primary (district) district cooling side";
+  parameter Modelica.SIunits.MassFlowRate m2_flow_nominal(min=0,start=0.5)
+    "Nominal mass flow rate of secondary (building) district cooling side";
+
+  // Heat exchanger
+  parameter Modelica.SIunits.PressureDifference dp1_nominal(min=0,start=500)
+    "Nominal pressure difference on primary side"
+    annotation(Dialog(tab="Heat exchanger"));
+  parameter Modelica.SIunits.PressureDifference dp2_nominal(min=0,start=500)
+    "Nominal pressure difference on secondary side"
+    annotation(Dialog(tab="Heat exchanger"));
+  parameter Boolean use_Q_flow_nominal=true
+    "Set to true to specify Q_flow_nominal and temperatures, or to false to specify effectiveness"
+    annotation(Dialog(tab="Heat exchanger"));
+  parameter Modelica.SIunits.HeatFlowRate Q_flow_nominal(min=0,start=10000)
+    "Nominal heat transfer"
+    annotation(Dialog(tab="Heat exchanger"));
+  parameter Modelica.SIunits.Temperature T_a1_nominal(min=0+273,max=100+273,start=5+273)
+    "Nominal temperature at port a1"
+    annotation(Dialog(tab="Heat exchanger"));
+  parameter Modelica.SIunits.Temperature T_a2_nominal(min=0+273,max=100+273,start=7+273)
+    "Nominal temperature at port a2"
+    annotation(Dialog(tab="Heat exchanger"));
+  parameter Modelica.SIunits.Efficiency eta(min=0,max=1)=0.8
+    "Constant effectiveness"
+    annotation(Dialog(tab="Heat exchanger"));
+
+  // Primary supply control valve
+  parameter Modelica.SIunits.PressureDifference dpValve_nominal(min=0)=50
+    "Control valve: Nominal pressure drop of fully open valve";
+
+  // Secondary supply pump
+  parameter Modelica.SIunits.PressureDifference dp_nominal(min=0)
+    "Secondary pump: Nominal pressure raise, used for default pressure curve if not specified in record per";
+
+  // Controller parameters
+  parameter Modelica.Blocks.Types.SimpleController controllerType=
+    Modelica.Blocks.Types.SimpleController.PID
+    "Type of controller"
+    annotation(Dialog(tab="Controller"));
+  parameter Real k(min=0, unit="1") = 1
+    "Gain of controller"
+    annotation(Dialog(tab="Controller"));
+  parameter Modelica.SIunits.Time Ti(min=Modelica.Constants.small)=0.5
+    "Time constant of integrator block"
+     annotation (Dialog(tab="Controller"));
```

Comment:
add the `enable` attribute to define when the parameter could be modified. This should be added to other parameters, `Td`, `wd`, `Ni`, `Nd`, `xi_start`, `xd_start`, `yCon_start`.

Reply:
Added `enable` to relevant parameters